src/libstd/sys/mod.rs

```diff
@@ -41,7 +41,7 @@ cfg_if::cfg_if! {
     } else if #[cfg(target_os = "wasi")] {
         mod wasi;
         pub use self::wasi::*;
-    } else if #[cfg(target_arch = "wasm32")] {
+    } else if #[cfg(any(target_arch = "wasm32", target_arch = "asmjs"))] {
```

Comment:
I was a bit surprised that things still worked with this, but I think that this is actually never hit. Since emscripten falls into the unix bucket above it's probably fine to ignore this

Reply:
Ah right, I didn't think about the Unix bucket. I'll remove "asmjs" here since it has no effect.